lib/logstash/outputs/elasticsearch_java.rb

```diff
@@ -61,14 +64,14 @@
 # - Events from the retry queue are submitted again either when the queue reaches its max size or when
 #   the max interval time is reached, which is set in :retry_max_interval.
 # - Events which are not retryable or have reached their max retry count are logged to stderr.
-class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
+class LogStash::Outputs::ElasticSearchJava < LogStash::Outputs::Base
   attr_reader :client
 
   include Stud::Buffer
-  RETRYABLE_CODES = [429, 503]
+  RETRYABLE_CODES = [409, 429, 503]
```

Comment:
Aha, from this commit from @Da-Wei https://github.com/andrewvc/logstash-output-elasticsearch_java/commit/781206496780ba5d7aac5bc7d518e4ddf435e602